vignettes/sablefish_example.Rmd

```diff
@@ -0,0 +1,178 @@
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+Working through PacFIN data can be complex. This examples guides you through working up the data for sablefish. The fish ticket data contains information on catches and is labeled `catch.pacfin` within this vignette. The biological data that contains information on lengths and ages is labeled `bds.pacfin`. These names are standard for every data set that is downloaded and saved from PacFIN.
+
+Before working up any of the data, a few items need to be clarified. Such as what gear types you would like to use and what bins should be used for the biological data. As well as which files you would like to load.
+
+```{r setup}
+# Species
+common_name <- "sablefish"
+species_code <- "SABL"
+
+# Files
+file_bds <- fs::path("..", "PacFIN.SABL.bds.11.Nov.2024.RData")
+file_catch <- fs::path("..", "PacFIN.SABL.CompFT.11.Nov.2024.RData")
+
+# Bins for biological data
+dolbins <- seq(18, 90, by = 2)
+doabins <- seq(0, 60, by = 1)
+
+# Gear codes
+used_gears <- c("HKL", "POT", "TWL")
+
+# Additional settings
+expansion <- 0.95
+good_lengths <- c("", "A", "F")
+good_methods <- "R"
+good_samples <- c("", "M", "C")
+good_states <- c("WA", "OR", "CA")
+```
+
+```{r read-data}
+# catch.pacfin
+# PullCatch.PacFIN(pacfin_species_code = species_code)
+load(file_catch)
+# bds.pacfin
+# PullBDS.PacFIN(pacfin_species_code = species_code)
+load(file_bds)
+```
+
+## Calculate the weight--length relationship
+
+A choice must be made about which data set you want to fit for estimating the weight--length relationships by sex and across sexes. Many scientists choose to use the survey data because it is thought to best represent the population. You could use the data retrieved from PacFIN to calculate this relationship but that option is not shown here.
+
+```{r weight-length-data, message = FALSE, warning = FALSE}
+bds_survey <- nwfscSurvey::pull_bio(
+  common_name = common_name,
+  survey = "NWFSC.Combo"
+)
+weight_length_estimates <- nwfscSurvey::estimate_weight_length(
+  bds_survey,
+  verbose = FALSE
+)
+```
+
+```{r print-weight-length-table}
+knitr::kable(weight_length_estimates, "markdown")
+```
+
+## Catch data
+
+The catch data must be summarized because it will be used in the analysis of the biological data. Where, `formatCatch()` takes a long data frame as input and converts it to a wide data frame with one row per year and one column per catch group. In the `r common_name` example the catch is stratified by `state` and `geargroup`, leading to one column for each state-gear combination. Catches are summarized in pounds because the weighting is done in pounds.
+
+```{r catch}
+catch_formatted <- catch.pacfin |>
+  getGearGroup(spp = common_name) |>
+  getState() |>
+  formatCatch(
+    strat = c("state", "geargroup"),
+    # TODO: Determine if we want ROUND_WEIGHT_LBS
+    valuename = "LANDED_WEIGHT_LBS"
+  )
```

Comment:
For most species round weight and landed weight are equal, but generally one should always used round weight.  Sablefish is one of the species where this matters.  Vessels often remove sablefish heads at-sea (or do some form of dressing reducing fish weight) and the round weight column is based on the landed weight of the partial fish multiplied by an expansion factor to estimate the weight of the whole fish.

Question - why are we using the pounds column?  I typically use the ROUND_WEIGHT_MTONS column to avoid having to do additional calculations.  Since the landed weight is used in the second stage expansion, scaling up samples by relative landed weight by gear and state, the units likely don't matter but I would have expected metric tons here.  Have I been doing it wrong?

Reply:
The length-weight relationship parameters are in units of lbs not mt, which is why I convert I think.

Reply:
Thanks.  This prompted me to dig into the code a bit further.  I modified the vignette some to point out to users that they should use an externally processed catch file if they have composition data prior to 1981 (prior to 1986 for OR) since PacFIN catches would not be available for all bds years in that case.  I also remembered that the 2nd stage expansion function has a `Units` argument, which is then used to determine if the catches need to be converted to pounds from metric tons which is handy if the user does not want to convert already processed catches into pounds themselves.